src/vm/arm/asmhelpers.S

```diff
@@ -509,6 +509,29 @@ LOCAL_LABEL(UM2MThunk_WrapperHelper_ArgumentsSetup):
         NESTED_END ThePreStub, _TEXT
 
 // ------------------------------------------------------------------
+        NESTED_ENTRY ThePreStubCompactARM, _TEXT, NoHandler
+
+        // r12 - address of compact entry point + PC_REG_RELATIVE_OFFSET
+
+        PROLOG_WITH_TRANSITION_BLOCK
+
+        mov         r0, r12
+
+        bl          C_FUNC(PreStubGetMethodDescForCompactEntryPoint)
+
+        mov         r1, r0                                          // pMethodDesc
+
+        add         r0, sp, #__PWTB_TransitionBlock                 // pTransitionBlock
+
+        bl          C_FUNC(PreStubWorker)
```

Comment:
It think it would be better to tailcall to ThePrestub instead - it saves you from the extra plumbing required for debugger here. It is what we do in other similar places, e.g. check PrecodeFixupThunk.